sbin/setkey/setkey.8

```diff
@@ -640,22 +645,22 @@ in the
 of the
 .Ar protocol
 parameter:
-.Bd -literal -offset indent
-algorithm	keylen (bits)	comment
-hmac-sha1	160		ah/esp: rfc2404
-		160		ah-old/esp-old: 128bit ICV (no document)
-null		0 to 2048	for debugging
-hmac-sha2-256	256		ah/esp: 128bit ICV (RFC4868)
-		256		ah-old/esp-old: 128bit ICV (no document)
-hmac-sha2-384	384		ah/esp: 192bit ICV (RFC4868)
-		384		ah-old/esp-old: 128bit ICV (no document)
-hmac-sha2-512	512		ah/esp: 256bit ICV (RFC4868)
-		512		ah-old/esp-old: 128bit ICV (no document)
-aes-xcbc-mac	128		ah/esp: 96bit ICV (RFC3566)
-		128		ah-old/esp-old: 128bit ICV (no document)
-tcp-md5		8 to 640	tcp: rfc2385
-chacha20-poly1305	256	ah/esp: 128bit ICV (RFC7634)
-.Ed
+.Bl -column -offset indent "chacha20-poly130" "Key Size" "ICV" "Comment"
```

Comment:
Unless you're an expert on cryptography, please don't change any text in this table.  Especially when (after reducing the offset) it fits into the standard 80-char-wide terminal.

"keylen (bits)" is much more specific than "Key Size".  If you moved "(bits)" onto the row below, I wouldn't complain (I suggested that, after all), but I don't recommending removing "bits" entirely.

The only text I'd feel comfortable removing are the "(no document)" labels.

Reply:
I'm not an expert in cryptography, and of course changing any text here would need someone who is to review even if I thought I was, but would you be willing to tell me more about this?

Googling, everything I found for keylen said it was an abreviation for key length, also called key size, and only given in bits.

Reply:
I'm extremely allergic to touching anything cryptographic.

Linguistically, I would say that "length" is a measurement of a 1-dimensional object; whereas "size" could be a measurement of a 2d or 3d object as well.  ("What's the size of that rectangle?")

Technically... well, for example, looking at the document which describes sha256, NIST-FIPS 180-4 [1], there's some places where they use "size", and other places where they use "length".  I'm not certain that there's a difference, but I'm also not certain that there isn't a difference.  So I personally would refrain from changing any current use of those terms in FreeBSD.

[1] https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf

Reply:
Removing the offset is always a last resort for improving legibility. I really appreciate the warning and have the same reservation. It doesn't mean we shouldn't respectfully try, it just means we can't pull it off alone. If I was a comitter, I would not commit this (or anything which changes content e.g. bumps date) until a SME comes to help. Eventually, one will get bored and look at this.